loginas/utils.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from django.conf import settings as django_settings
+from django.contrib.auth.models import User
+from django.contrib.auth import load_backend, login, logout
+from django.contrib import messages
+from django.core.signing import TimestampSigner, SignatureExpired
+from datetime import timedelta
+
+from . import settings
+
+signer = TimestampSigner()
+
+logger = logging.getLogger(__name__)
+
+
+def login_as(user, request, store_original_user=True):
+    """
+    Utility function for forcing a login as specific user -- be careful about
+    calling this carelessly :)
+    """
+
+    # Save the original user pk before it is replaced in the login method
+    original_user_pk = request.user.pk
+
+    # Find a suitable backend.
+    if not hasattr(user, 'backend'):
+        for backend in django_settings.AUTHENTICATION_BACKENDS:
+            if user == load_backend(backend).get_user(user.pk):
+                user.backend = backend
+                break
+
+    # Log the user in.
+    if hasattr(user, 'backend'):
+        login(request, user)
+
+    # Set a flag on the session
+    if store_original_user:
+        messages.warning(request, "Your login is now switched to {} - log out to become your original user!".format(user.username))
+        request.session[settings.USER_SESSION_FLAG] = signer.sign(original_user_pk)
+
+
+def restore_original_login(request):
+    """
+    Restore an original login session, checking the signed session
+    """
+
+    original_session = request.session.get(settings.USER_SESSION_FLAG)
+    logout(request)
+
+    if not original_session:
+        return
+
+    try:
+        original_user_pk = signer.unsign(
+            original_session,
+            max_age=timedelta(days=2)
```

Comment:
That's on purpose, it's the best way of displaying the age in a readable way. `unsign` does this:

```
            if isinstance(max_age, datetime.timedelta):
                max_age = max_age.total_seconds()
```